include/PriorityQueue.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <iostream>
+#include <list>
+#include <exception>
+
+template<class T>
+struct MyComparator {
+	T operator()(T a, T b) const {
+		return a - b;

```

Comment:
Notice that marking the arguments simply as T means they will be copied every call to the comparator. This is bad for performance and sometimes even impossible (i.e - if T is or contains a unique_ptr or any other type that deleted the Copy-constructor).
So in this case it’s definitely better to use "const-reference".
Check it out: https://www.geeksforgeeks.org/difference-between-call-by-value-and-call-by-reference/